cpp-lib/src/pitaya/utils.cpp

```diff
@@ -64,11 +64,10 @@ ParseEtcdKey(const string& key,
     }
 
     auto comps = string_utils::Split(key, '/');
-
     auto prefix = comps[0];
-    
-    // Fail if the key does not start with "<etcdPrefix>/"
-    if ((prefix + "/") != etcdPrefix) {
+
+    // Fail if key does not start with "<etcdPrefix>/"
+    if (key.find(etcdPrefix) != 0 || etcdPrefix.find('/') == std::string::npos) {
```

Comment:
In this function, we still assume that the key has a `servers` inside it. However, the user now can give a prefix that does not contain `servers`, so maybe we have to think about this use case.

Reply:
but it is assumed that key must have a set of components (servers/serverType/serverID)